httpc.go

```diff
@@ -298,14 +299,60 @@ func (r *Request) Run() error {
 	return r.RunWithContext(context.Background())
 }
 
+var (
+	errorEncodedRawBodyCollision = errors.New("cannot use both body encoding and raw body content")
+)
+
+func (r *Request) prepBody() (body io.ReadCloser, err error) {
+	// If requested, parse the requst body using the specified encoder
```

Comment:
Trivial nitpick: Please add an empty line before the comment to be consistent with the rest of the formatting.